plugins/modules/adguardhome_rewrite.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2025 Markus Bergholz
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r"""
+module: adguardhome_rewrite
+
+short_description: Add, update or delete DNS rewrite rules from AdGuard Home
+
+version_added: 3.3.0
+
+description:
+  - Add, update or delete DNS rewrite rules from AdGuard Home.
+extends_documentation_fragment:
+  - community.dns.adguardhome.connectivity
```

Comment:
All modules should use the standard attributes doc fragment:
```suggestion
  - community.dns.adguardhome.connectivity
  - community.dns.attributes
```